src/zjs_spi.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017, Intel Corporation.
+
+#ifndef __zjs_spi_h__
+#define __zjs_spi_h__
+
+#include "jerryscript.h"
+
+enum SPITopology {
+    ZJS_TOPOLOGY_FULL,
+    ZJS_TOPOLOGY_SINGLE_READ,
+    ZJS_TOPOLOGY_SINGLE_WRITE,
+    ZJS_TOPOLOGY_MULTIPLEXED,
+    ZJS_TOPOLOGY_DAISY_CHAIN
+};
+
+enum direction {
+    ZJS_SPI_DIR_UNDEFINED,
```

Comment:
Originally I had intended this to be for when a direction wasn't passed in as an arg.  However, its not really necessary so I'll remove it to keep things clear.